src/components/ChooseDialect.js

```diff
@@ -31,25 +32,33 @@ const ChooseDialect = ({ navigation, dispatch, dialects }) => {
           counter = 0;
         }, 1200);
       } else if (counter === 2) {
-        navigation.navigate("PatternList", { grammars });
+        navigation.navigate("PatternList", { grammars, id });
       }
     } else {
       setSelectedDialectId(id);
       counter = 1;
     }
   };
 
+  const changeBtnText = (id) => {
+    const selectedDialect = dialects.find((dialect) => dialect.id === id);
+    setBtnText(selectedDialect.start_btn_text);
+  };
+
   const navigateBtn = (id) => {
     if (id) {
       const selectedDialect = dialects.find((dialect) => dialect.id === id);
-      navigation.navigate("PatternList", { grammars: selectedDialect.grammars });
+      navigation.navigate("PatternList", {
+        grammars: selectedDialect.grammars,
+        id
```

Comment:
I would pass in the `selectedDialect` so you can get both `selectedDialect.grammars` and `selectedDialect. next_btn_text` in the src/components/PatternList.js